tests/client/components/carousel.spec.js

```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
 
-    expect(wrapper.vm.carouselWidth).toBe(0);
+    it.skip('should navigate to slide 2', () => {});
 
-    expect(wrapper).toMatchSnapshot();
+    // TODO: Is this desired functionality?
+    it.skip('should navigate to slide to even when navigation is not enabled', () => {});
   });
 
-  it('should go to second slide when we have odd number of slides and recompute carousel width', done => {
-    const wrapper = mount(Carousel, {
-      propsData: {
-        scrollPerPage: true,
-        perPage: 2
-      },
-      slots: {
-        default: [Slide, Slide, Slide, Slide, Slide]
-      }
+  describe('Carousel pagination', () => {
+    it.skip('should render the pagination component when pagination is enabled', () => {});
+
+    it.skip('should not render the pagination component when pagination is disabled', () => {});
+
+    // TODO: It seems that the pagination component is heavily connected to the carousel, decouple this?
+    // If no then we should add tests here in the carousel for it
+  });
+
+  describe('Carousel slides per page', () => {
+    it('should apply custom slides per page when responsive param provided', () => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          perPageCustom: [[0, 20]]
+        }
+      });
+
+      expect(wrapper.vm.currentPerPage).toBe(20);
+
+      expect(wrapper).toMatchSnapshot();
     });
-    wrapper.vm.carouselWidth = 500;
 
-    wrapper.vm.$nextTick(() => {
-      wrapper.vm.goToPage(1);
-      wrapper.vm.computeCarouselWidth();
+    it('should fall back to default slides per page when no responsive param provided', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          scrollPerPage: true,
+          perPageCustom: [[9999, 20]]
+        },
+        slots: {
+          default: [Slide, Slide, Slide]
+        }
+      });
 
-      expect(wrapper.vm.currentPage).toBe(1);
+      expect(wrapper.vm.currentPerPage).toBe(2);
+      expect(wrapper.vm.pageCount).toBe(2);
 
-      done();
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
     });
-  });
 
-  it('should register 0 slides when 0 slides are added to the slots', () => {
-    const wrapper = mount(Carousel);
+    it.skip('should have 2 slides on the current page by default', () => {});
 
-    expect(wrapper.vm.slideCount).toBe(0);
+    it.skip('should have 3 slides on the current page when perPage is set to 3', () => {});
 
-    expect(wrapper).toMatchSnapshot();
+    it.skip('should have 3 slides when perPage is set to 6', () => {});
+
+    it.skip('should only have one page when number of slides is less than slides per page', () => {});
   });
 
-  it('should register 3 slides when 3 slides are added to the slots', done => {
-    const wrapper = mount(Carousel, {
-      slots: {
-        default: [Slide, Slide, Slide]
-      }
+  describe('Resistance co-efficient', () => {
```

Comment:
Just making a note here to have a better name for this, it's internal only right?

Reply:
This is a prop you pass in, I tried to make it more clear but it's hard to describe this in a test...